concent_api/core/message_handlers.py

```diff
@@ -519,18 +525,36 @@ def handle_send_force_subtask_results(client_message: message.concents.ForceSubt
             reason=message.concents.ForceSubtaskResultsRejected.REASON.RequestPremature,
         )
 
-    subtask = store_or_update_subtask(
-        task_id                     = task_to_compute.compute_task_def['task_id'],
-        subtask_id                  = task_to_compute.compute_task_def['subtask_id'],
-        provider_public_key         = provider_public_key,
-        requestor_public_key        = requestor_public_key,
-        state                       = Subtask.SubtaskState.FORCING_ACCEPTANCE,
-        next_deadline               = forcing_acceptance_deadline + settings.CONCENT_MESSAGING_TIME,
-        set_next_deadline           = True,
-        ack_report_computed_task    = client_message.ack_report_computed_task,
-        task_to_compute             = client_message.ack_report_computed_task.report_computed_task.task_to_compute,
-        report_computed_task        = client_message.ack_report_computed_task.report_computed_task,
+    subtask = get_one_or_none_subtasks_from_database(
+        subtask_id=task_to_compute.compute_task_def['subtask_id'],
     )
+
+    if subtask is None:
+        subtask = store_subtask(
+            task_id=task_to_compute.compute_task_def['task_id'],
+            subtask_id=task_to_compute.compute_task_def['subtask_id'],
+            provider_public_key=provider_public_key,
+            requestor_public_key=requestor_public_key,
+            state=Subtask.SubtaskState.FORCING_ACCEPTANCE,
+            next_deadline=forcing_acceptance_deadline + settings.CONCENT_MESSAGING_TIME,
+            task_to_compute=client_message.ack_report_computed_task.report_computed_task.task_to_compute,
+            report_computed_task=client_message.ack_report_computed_task.report_computed_task,
+            ack_report_computed_task=client_message.ack_report_computed_task,
+        )
+    else:
+        if subtask.state_enum == Subtask.SubtaskState.FORCING_ACCEPTANCE:
+            return message.concents.ServiceRefused(
+                reason=message.concents.ServiceRefused.REASON.DuplicateRequest,
+            )
+        subtask = update_subtask(
+            subtask=subtask,
+            state=Subtask.SubtaskState.FORCING_ACCEPTANCE,
+            next_deadline=forcing_acceptance_deadline + settings.CONCENT_MESSAGING_TIME,
```

Comment:
I think `update` suggests` function gets subtask, makes something with it-updated and returns. I need subtask returned to pass it to `store_pending_message`. It can be done without passing subtask, but in `store_pending_message` it had to been received from database. I don't think we want it.. 

If it is not obvious maybe leave the code as it is now and comment it.

Reply:
What is mean is that the function does not need to return `subtask` to update it. Returning it in addition to updating it only makes it confusing because that's not a typical construct.

Reply:
If it is not typical I don't discuss any more. As you said at the beginning we will repair it in separately. For now as a quick repair i changed the name to `update_and_return_updated_subtask`. It is not nice name but shows that subtask is also returned.. Is it OK?

Reply:
It's OK for now.

But I think we should change it eventually. This function should not return anything.